### PR TITLE
enh(core): replace centcore with gorgoned

### PR DESCRIPTION
**Pull request by sc979**
## Description

Replace centcore with gorgone
Add ZMQ options and new fields in the configuration
Manage transparent update for platforms linked to remote/poller
Upgrade the source installation script

**Fixes** # (none)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

download the new files, stop centore.service and upgrade to 20.04
start gorgoned.service and check that everything run as expected in the IHM

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [x] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [x] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: src/Centreon/Domain/ConfigurationLoader/YamlConfigurationLoader.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
+<?php
+/*
```
Comment (kduret) [inline src/Centreon/Domain/ConfigurationLoader/YamlConfigurationLoader.php]:
```suggestion

/*
```

File: src/Centreon/Domain/ConfigurationLoader/YamlConfigurationLoader.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+class YamlConfigurationLoader
+{
+    private const INCLUDE_TOKEN = 'include';
+
+    /**
+     * @var string Root configuration file
+     */
+    private $configurationFile;
+
+    public function __construct(string $configurationFile)
```
Comment (kduret) [inline src/Centreon/Domain/ConfigurationLoader/YamlConfigurationLoader.php]: describe params

File: src/Centreon/Domain/Gorgone/Interfaces/GorgoneServiceInterface.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\Gorgone\Interfaces;
+
+interface GorgoneServiceInterface
```
Comment (kduret) [inline src/Centreon/Domain/Gorgone/Interfaces/GorgoneServiceInterface.php]: describe purpose of this interface

File: src/Centreon/Domain/Gorgone/Interfaces/ResponseRepositoryInterface.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\Gorgone\Interfaces;
+
+interface ResponseRepositoryInterface
```
Comment (kduret) [inline src/Centreon/Domain/Gorgone/Interfaces/ResponseRepositoryInterface.php]: describe purpose of this interface

File: src/Centreon/Domain/Option/Interfaces/OptionRepositoryInterface.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\Option\Interfaces;
+
+use Centreon\Domain\Option\Option;
+
+interface OptionRepositoryInterface
```
Comment (kduret) [inline src/Centreon/Domain/Option/Interfaces/OptionRepositoryInterface.php]: describe purpose of this interface

File: src/Centreon/Domain/Option/Interfaces/OptionServiceInterface.php [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\Option\Interfaces;
+
+use Centreon\Domain\Option\Option;
+
+interface OptionServiceInterface
```
Comment (kduret) [inline src/Centreon/Domain/Option/Interfaces/OptionServiceInterface.php]: describe purpose of this interface

File: src/Centreon/Domain/Option/Option.php [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\Option;
+
+class Option
```
Comment (kduret) [inline src/Centreon/Domain/Option/Option.php]: describe purpose of this class

File: src/Centreon/Infrastructure/Gorgone/CommandRepositoryAPI.php [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+     */
+    private $configuration;
+
+    /**
+     * @param OptionServiceInterface $optionService
+     * @param ConfigurationLoaderApiInterface $configuration
+     */
+    public function __construct(OptionServiceInterface $optionService, ConfigurationLoaderApiInterface $configuration)
+    {
+        $this->client = new CurlHttpClient();
```
Comment (kduret) [inline src/Centreon/Infrastructure/Gorgone/CommandRepositoryAPI.php]: could you use dependency injection ?

File: src/Kernel.php [hunk 9]
```diff
@@ -42,6 +45,21 @@ class Kernel extends BaseKernel
 
     private const CONFIG_EXTS = '.{php,xml,yaml,yml}';
 
+    public static function createForWeb(): Kernel
```
Comment (kduret) [inline src/Kernel.php]: add phpdoc

File: tests/php/Centreon/Domain/ConfigurationLoader/YamlConfigurationLoaderTest.php [hunk 10]
```diff
@@ -0,0 +1,87 @@
+<?php
+/*
```
Comment (kduret) [inline tests/php/Centreon/Domain/ConfigurationLoader/YamlConfigurationLoaderTest.php]:
```suggestion

/*
```

File: tests/php/Centreon/Domain/Gorgone/CommandServiceTest.php [hunk 11]
```diff
@@ -0,0 +1,81 @@
+<?php
+/*
```
Comment (kduret) [inline tests/php/Centreon/Domain/Gorgone/CommandServiceTest.php]:
```suggestion

/*
```

File: www/class/centreon-clapi/centreonInstance.class.php [hunk 12]
```diff
@@ -127,6 +138,9 @@ public function initUpdateParameters($parameters)
         }
 
         $objectId = $this->getObjectId($params[self::ORDER_UNIQUENAME]);
+        if ($params[1] === 'gorgone_communication_type') {
+            $params[2] =  self::GORGONE_COMMUNICATION[$params[2]];
```
Comment (kduret) [inline www/class/centreon-clapi/centreonInstance.class.php]:
```suggestion
            $params[2] = self::GORGONE_COMMUNICATION[$params[2]];
```

File: www/include/Administration/parameters/gorgone/help.php [hunk 13]
```diff
@@ -0,0 +1,51 @@
+<?php
+$help = array();
```
Comment (kduret) [inline www/include/Administration/parameters/gorgone/help.php]:
```suggestion
$help = [];
```

File: doc/fr/api/clapi/objects/instances.rst [hunk 14]
```diff
@@ -23,29 +23,31 @@ Columns are the following:
...
-Activate	  *1* if poller is enabled, *0* otherwise
+Activate          *1* if poller is enabled, *0* otherwise
 
-Status	          *1* if poller is running, *0* otherwise
+Status            *1* if poller is running, *0* otherwise
 
-Init script	  Init script path
+Init script       Init script path
 
 Monitoring Engine Engine used on poller: *NAGIOS*, *ICINGA*, *SHINKEN*...
```
[the remark this answers is not in the document]
Comment (sc979) [reply doc/fr/api/clapi/objects/instances.rst]: the documentation needs to be re-writted ;) some part are sooo old

File: logrotate/centcore [hunk 15]
```diff
@@ -1,10 +0,0 @@
-@CENTREON_LOG@/centcore.log {
```
[the remark this answers is not in the document]
Comment (sc979) [reply logrotate/centcore]: it's managed from the gorgone module directly. As the module needs to stay independent from centreon :
https://github.com/centreon/centreon-gorgone/blob/master/config/logrotate/gorgoned

File: install.sh [hunk 16]
```diff
@@ -219,6 +210,21 @@ for binary in $BINARIES; do
 	fi
 done
 
+###### Mandatory step
+# ask if gorgone have already been installed
```
Comment (sc979) [inline install.sh]:
```suggestion
# ask if gorgone is already installed
```

**Comment by sc979** [comment]
WARNING -> this PR require the centreon-build's one to be merged too to avoid broking the CI 
https://github.com/centreon/centreon-build/pull/239

**Pull request merged**